pkg/platform/interface.go

```diff
@@ -1,7 +1,23 @@
 package platform
 
-import "io"
+import (
+	"io"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
 
 type Client interface {
 	ExecCMDInContainer(containerName, podName string, cmd []string, stdout io.Writer, stderr io.Writer, stdin io.Reader, tty bool) error
+
+	GetPodLogs(podName, containerName string, followLog bool) (io.ReadCloser, error)
+
+	GetPodsMatchingSelector(selector string) (*corev1.PodList, error)
+
+	// GetAllResourcesFromSelector returns all resources of any kind (including CRs) matching the given label selector
+	GetAllResourcesFromSelector(selector string, ns string) ([]unstructured.Unstructured, error)
+
+	GetAllPodsInNamespaceFromSelector(selector string, ns string) (*corev1.PodList, error)
+
+	GetRunningPodFromSelector(selector string) (*corev1.Pod, error)
```

Comment:
Should the comments be expected here or at the places where the interface is being implemented?

Reply:
I've added a few (implementation-independent) comments here as well.

Reply:
That's indeed what we should have in the interface. Implementation specific comment should go into the package/file where it's getting implemented.

Thanks for adding the comments. 👍🏾